### PR TITLE
fix(app): handle dtwiz after estop

### PR DESCRIPTION
**Pull request by TamarZanzouri**
# Overview

Fixes some issues around estop and drop tip. Specifically,
- When you hit the estop, we surface an error and lose positioning - normal for errors like this - but we also turn off the motors, and they need to be turned back on, so add a command to do that and use it in dtwiz
- on desktop, the estop takeover wouldn't work if you were anywhere but the top level devices page because of either the react-router update or because of an oversight that was made during the flex launch
- the estop takeover is polling, and slowly. we should add notifications at some point but for now we can do some strategic query invalidations and fast polling when the estop is triggered to make it more responsive

Closes RQA-3103, RQA-3133

## Test Plan and Hands on Testing

- run a protocol, stop it with estop, and check that the modal fires; is responsive; and leaves you to a dtwiz, which opertaes correctly
  - [x] on desktop, viewing the run, while network connected
  - [x] on desktop, viewing the run, while usb connected
  - [x] on odd

### CODE REVIEW

File: app/src/App/DesktopApp.tsx [hunk 1]
```diff
@@ -158,11 +158,10 @@ export const DesktopApp = (): JSX.Element => {
 }
 
 function RobotControlTakeover(): JSX.Element | null {
-  const deviceRouteMatch = useMatch('/devices/:robotName')
-  const params = deviceRouteMatch?.params as DesktopRouteParams
-  const robotName = params?.robotName
+  const deviceRouteMatch = useMatch('/devices/:robotName/*')
```
Comment (mjhuff) [inline app/src/App/DesktopApp.tsx:161]: Super funny/weird change between RR versions.

It looks like this is the only place in the app we use `useMatch`.

File: app/src/organisms/EmergencyStop/EstopTakeover.tsx [hunk 2]
```diff
@@ -14,16 +14,34 @@ import {
...
 export function EstopTakeover({ robotName }: EstopTakeoverProps): JSX.Element {
+  const [estopEngaged, setEstopEngaged] = React.useState<boolean>(false)
+  const [
+    isWaitingForLogicalDisengage,
+    setIsWaitingForLogicalDisengage,
+  ] = React.useState<boolean>(false)
   const { data: estopStatus } = useEstopQuery({
-    refetchInterval: ESTOP_REFETCH_INTERVAL_MS,
+    refetchInterval: estopEngaged
+      ? ESTOP_CURRENTLY_ENGAGED_REFETCH_INTERVAL_MS
```
Comment (mjhuff) [inline app/src/organisms/EmergencyStop/EstopTakeover.tsx:32]: Yeah, great thinking.

File: app/src/organisms/EmergencyStop/EstopTakeover.tsx [hunk 3]
```diff
@@ -14,16 +14,34 @@ import {
...
+    onSuccess: response => {
+      setEstopEngaged(
+        [PHYSICALLY_ENGAGED || LOGICALLY_ENGAGED].includes(
+          response?.data.status
+        )
+      )
+      setIsWaitingForLogicalDisengage(false)
+    },
   })
+  console.log(`estop status: ${estopStatus?.data.status}`)
```
Comment (mjhuff) [inline app/src/organisms/EmergencyStop/EstopTakeover.tsx]: stray log?

File: app/src/organisms/EmergencyStop/EstopPressedModal.tsx [hunk 4]
```diff
@@ -155,14 +177,19 @@ function DesktopModal({
...
+    acknowledgeEstopDisengage(
+      {},
+      {
+        onSuccess: () => {
+          setShouldSeeLogicalDisengage()
+          closeModal()
+        },
+        onError: (error: any) => {
+          setIsResuming(false)
+          console.log(error)
```
Comment (mjhuff) [inline app/src/organisms/EmergencyStop/EstopPressedModal.tsx]: maybe `console.error`?

**Review by mjhuff — APPROVED** [review]
This was surprisingly involved. Thanks to you both for fixing this!

File: app/src/organisms/EmergencyStop/EstopTakeover.tsx (hunk 3 above)
Comment (sfoster1) [reply app/src/organisms/EmergencyStop/EstopTakeover.tsx]: yup, good catch, thanks!

**Pull request merged**